gear-test-lib/src/system.rs

```diff
@@ -0,0 +1,93 @@
+use codec::Encode;
+use env_logger::{Builder, Env};
+use gear_core::{message::Message, program::ProgramId};
+use std::{fmt::Debug, sync::Mutex};
+
+use crate::{manager::ExtManager, program::ProgramIdWrapper};
+
+pub struct System(pub(crate) Mutex<ExtManager>);
+
+impl Default for System {
+    fn default() -> Self {
+        Self(Mutex::new(ExtManager::new()))
+    }
+}
+
+use colored::Colorize;
+
+use std::io::Write;
+
+impl System {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn init_logger(&self) {
+        let _ = Builder::from_env(Env::default().filter_or("gwasm", "gwasm=debug"))
+            .format(|buf, record| {
+                writeln!(
+                    buf,
+                    "[{}] {}",
+                    record.level().to_string().blue(),
+                    record.args().to_string().replacen("DEBUG: ", "", 1).white()
+                )
+            })
+            .format_target(false)
+            .format_timestamp(None)
+            .try_init();
+    }
+
+    pub fn send_message(&self, message: Message) {
+        self.0.lock().unwrap().run_message(message)
+    }
+
+    pub fn set_user<I: Into<ProgramIdWrapper> + Clone + Debug>(&self, user: I) {
+        let program_id: ProgramId = user.clone().into().into();
+
+        let mut system = self.0.lock().unwrap();
+
+        if system.programs.contains_key(&program_id) {
+            panic!(
+                "Can't set user {:?}, because Program with this id already exists",
+                user
+            )
+        }
+
+        system.user = program_id;
+    }
+
+    pub fn assert_user<I: Into<ProgramIdWrapper> + Clone + Debug>(&self, user: I) {
+        let program_id: ProgramId = user.clone().into().into();
+
+        if self.0.lock().unwrap().user != program_id {
+            panic!("User {:?} isn't actual user", user)
+        }
+    }
+
+    pub fn get_user(&self) -> ProgramId {
+        self.0.lock().unwrap().user
+    }
+
+    pub fn assert_log<E: Encode>(&self, from: u64, payload: E) {
+        self.assert_log_bytes(from, payload.encode())
+    }
+
+    pub fn assert_log_bytes<T: AsRef<[u8]>>(&self, from: u64, payload: T) {
```

Comment:
this should use `assert!` internally

Reply:
Now custom assertions removed. Only native ones in tests